src/atcore.cpp

```diff
@@ -172,9 +182,12 @@ void AtCore::print(const QString &fileName)
     while (!gcodestream.atEnd()) {
         QCoreApplication::processEvents(); //make sure all events are processed.
         switch (state()) {
+        case STARTPRINT:
         case IDLE:
         case BUSY:
-            setState(BUSY);
+            if (state() != BUSY) {
```

Comment:
I have removed that line and done as suggested and made the check part of setState